test/junit/scala/tools/nsc/backend/jvm/opt/CallGraphTest.scala

```diff
@@ -150,6 +152,22 @@ class CallGraphTest extends ClearAfterClass {
     checkCallsite(dg1Call, t2, g1, cMClassBType, true, false, false)
   }
 
+  @Test
+  def callerSensitiveNotSafeToInline(): Unit = {
+    val code =
+      """class C {
+        |  def m = java.lang.Class.forName("C")
+        |}
+      """.stripMargin
+    val List(c) = compile(code)
+    val m = getMethod(c, "m")
+    val List(fn) = callsInMethod(m)
+    val forNameMeth = byteCodeRepository.methodNode("java/lang/Class", "forName", "(Ljava/lang/String;)Ljava/lang/Class;").get._1
+    val classTp = classBTypeFromInternalName("java/lang/Class")
+    val r = callGraph.callsites(m)(fn)
+    checkCallsite(fn, m, forNameMeth, classTp, safeToInline = false, atInline = false, atNoInline = false)
+  }
```

Comment:
This feature and test will only work on Open- and Oracle JDK, given our need to look for the `@sun.reflect.CallerSensitive`.

I guess we could make this test skip if `Class.forName("sun.reflect.CallerSensitive")` throws a `ClassNotFoundException`. But I'm sure there are plenty of other tests that would fail on other JDKs, so I don't think we need to bother here.